runtime/src/main/java/org/corfudb/runtime/view/stream/AddressMapStreamView.java

```diff
@@ -159,16 +160,31 @@ protected boolean discoverAddressSpace(final UUID streamId,
                 queue.addAll(streamAddressSpace.copyAddressesToSet(maxGlobal));
 
                 long trimMark = streamAddressSpace.getTrimMark();
+
+                // In case we are dealing with a stream that does not have the checkpoint
+                // capability, check to see if we are trying to access an address that has been
+                // previously trimmed.
+                if (!isCheckpointCapable()
+                        && Address.isAddress(trimMark)
+                        && trimMark > stopAddress) {
```

Comment:
Question here: Why we need to add a new `if` block here? I think the `isTrimCoveredByCheckpointOrLocalView(trimMark)` in the next `if` block covers the case of a non checkpoint capable stream, because it checks if `trimMark` is in the resolved queue, if it's not, it will throw a TrimmedException. The only difference I see is `options.ignoreTrimmed`, because in case of non checkpoint capable stream, we need to ignore this option and always throw TrimmedException. Then I think we don't need to add this `if` block, just add a isCheckpointCapable in the next `if` block when we check the option:

```java
if (isCheckpointCapable() && options.ignoreTrimmed) {              
    log.debug("getStreamAddressMap[{}]: Ignoring trimmed exception for address[{}].",
                                this, streamAddressSpace.getTrimMark());
 } else {
    log.warn(.....)
    throw new TrimmedException(message);
}
```

Is that true?

Reply:
Option ignoreTrimmed needs to go away since it was introduced to deal with trimmed exceptions at the beginning of the stream consumption life-cycle.